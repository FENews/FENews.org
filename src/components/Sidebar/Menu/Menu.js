import React from 'react';
import { Link } from 'gatsby';
import styles from './Menu.module.scss';

const Menu = ({ menu }) => (
  <nav className={styles['menu']}>
    <ul className={styles['menu__list']}>
      {menu.map((item) => (
        <li className={styles['menu__list-item']} key={item.path}>
          <Link
            to={item.path}
            className={styles['menu__list-item-link']}
            activeClassName={styles['menu__list-item-link--active']}
          >
            {item.label}
          </Link>
        </li>
      ))}
    </ul>
  </nav>
);

export default Menu;
