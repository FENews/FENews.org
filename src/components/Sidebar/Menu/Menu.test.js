import React from 'react';
import renderer from 'react-test-renderer';
import Menu from './Menu';

describe('Menu', () => {
  const props = {
    menu: [
      {
        label: 'Item 0',
        path: '/#0/'
      },
      {
        label: 'Item 1',
        path: '/#1/'
      }
    ]
  };

  it('renders correctly', () => {
    const tree = renderer.create(<Menu {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
