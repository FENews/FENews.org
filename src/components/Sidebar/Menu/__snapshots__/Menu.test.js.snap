// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu renders correctly 1`] = `
<nav
  className="menu"
>
  <ul
    className="menu__list"
  >
    <li
      className="menu__list-item"
    >
      <Link
        activeClassName="menu__list-item-link--active"
        className="menu__list-item-link"
        to="/#0/"
      >
        Item 0
      </Link>
    </li>
    <li
      className="menu__list-item"
    >
      <Link
        activeClassName="menu__list-item-link--active"
        className="menu__list-item-link"
        to="/#1/"
      >
        Item 1
      </Link>
    </li>
  </ul>
</nav>
`;
